src/components/button/Button.js

```diff
@@ -12,9 +12,8 @@ import PropTypes from 'prop-types';
  * @param {typography} props capitalize, uppercase, lowercase, only
  */
 const Button = (props) => {
-  const { children, type, size, shape, typography, disabled, fullWidth, id, onClick } = props;
+  const { children, type, size, shape, typography, disabled, fullWidth, id, onClick, className } = props;
```

Comment:
Is it possible not to pass `className` down? The reason for this is when people start creating their own classes and pass it down if we want to change the base class or extend it, it will break someone's implementation.

The react team were also saying they have seen that practice to make it hard later on because if you want change the base class that will inevitbly break someone's implementation

Reply:
It might be worth creating several types of buttons and pass that down. Similarly how they have it in [material design](https://material.io/design/components/buttons.html) i.e. outlined button, contained button etc

Reply:
What I mean is we have `type` prop so use that to determine what button we want to use and not modify it additionally.

Thoughts?

Reply:
We do have a `type, size, shape` to render different variations/styles. The `className` here is only to set margins, not really to update any of the button styles.

I do see the benefit of not allowing to add any style to components, but at the same time that will require us to work around that by an extra parent component and add the margins/spacing there.

```
<div className="flex">
  <div className="flex-1">
  </div>
  <div className="flex-1">
    <div className="mx-4">   //<---- this is the tradeoff, an extra div
      <Button />
     </div>
  </div>
</div>
```

Reply:
What about if we look at it like this:
* Each component should look the same regardless if where we display it
* The layout and margin etc should not be the responsibility of our component (i.e. buttons, inputs etc - they should all have default margin, padding etc and never change) but a responsibility of the layout/page.
 
Thoughts?

Reply:
Yeah, the tradeoff is fine for the benefits we get.